### PR TITLE
ref(autofix): Remove codebase indexing

### LINKED ISSUE #1011: (autofix): Refactor to remove unused codebase indexing code and logic

### PR DESCRIPTION
**Pull request by jennmueng**
Remove all unused functionality for codebase indexing for the foreseeable future in order to make maintaining this repo easier. **Removes almost 5k LoC!!!**

We may add it back in the future.

Leaving the codebase indexing status endpoint up until sentry side is clear.

Resolves #1011

### CODE REVIEW

File: src/migrations/versions/bd9b03d19b04_migration.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+"""Migration
```
Comment (roaga) [inline src/migrations/versions/bd9b03d19b04_migration.py:1]: is this file needed in the remote repo? Probably is, just making sure because I don't know what it does, and it seems somewhat auto-generated?

File: src/seer/app.py [hunk 2]
```diff
@@ -170,13 +143,8 @@ def repo_access_check_endpoint(data: RepoAccessCheckRequest) -> RepoAccessCheckR
...
 ) -> CodebaseStatusCheckResponse:
-    return CodebaseStatusCheckResponse(
-        status=get_codebase_index_status(
-            organization_id=data.organization_id,
-            project_id=data.project_id,
-            repo=data.repo,
-        )
-    )
+    # TODO: Remove this once sentry side is updated
+    return CodebaseStatusCheckResponse(status="up_to_date")
```
Comment (roaga) [inline src/seer/app.py:147]: I believe Sentry side has been updated by another PR that you merged in? Can we remove this now?

File: src/seer/automation/autofix/steps/steps.py [hunk 3]
```diff
@@ -60,7 +51,7 @@ def _get_extra_invoke_kwargs(self) -> dict[str, Any]:
                 "org_id": cur.request.organization_id,
                 "project_id": cur.request.project_id,
                 "group_id": group_id,
-                "codebase_indexing": not cur.request.options.disable_codebase_indexing,
+                "codebase_indexing": False,
```
Comment (roaga) [inline src/seer/automation/autofix/steps/steps.py:54]: Does this parameter need to be here at all anymore? Why just set to false?

File: src/seer/automation/codebase/models.py [hunk 4]
```diff
@@ -47,210 +22,11 @@ class Document(BaseDocument):
...
-    INDEXING = "indexing"
-    OUT_OF_DATE = "out_of_date"
-    NOT_INDEXED = "not_indexed"
-
-
 class RepoAccessCheckRequest(BaseModel):
     repo: RepoDefinition
 
 
+# TODO: Remove this once sentry side is updated
```
Comment (roaga) [inline src/seer/automation/codebase/models.py:29]: Isn't Sentry side updated now?

File: tests/test_seer.py [hunk 5]
```diff
@@ -458,6 +460,37 @@ def test_autofix_evaluation_start_endpoint_with_run_on_item_id(
...
+
+        # Assert that the response is correct
+        self.assertEqual(response.status_code, 200)
+        response_data = json.loads(response.get_data(as_text=True))
+        self.assertEqual(response_data, {"status": "up_to_date"})
+
+        # Assert that CodebaseStatusCheckResponse was called with the correct status
+        mock_response.assert_called_once_with(status="up_to_date")
+
+
```
Comment (roaga) [inline tests/test_seer.py:493]: Why do we need a new test related to codebase indexes? Aren't we removing those?

**Review by roaga — APPROVED** [review]
Very satisfying PR...

Some minor questions of more things we can remove before you merge below.

File: src/migrations/versions/bd9b03d19b04_migration.py (hunk 1 above)
Comment (jennmueng) [reply src/migrations/versions/bd9b03d19b04_migration.py:1]: Yes this folder is where we define migrations for our db

File: src/seer/app.py (hunk 2 above)
Comment (jennmueng) [reply src/seer/app.py:147]: that pr went in faster than I thought, but let's keep this here for a bit for the sake of safety, because without this we could error on every issue details page

File: src/seer/automation/codebase/models.py (hunk 4 above)
Comment (jennmueng) [reply src/seer/automation/codebase/models.py:29]: replied above

File: tests/test_seer.py (hunk 5 above)
Comment (jennmueng) [reply tests/test_seer.py:493]: to get codecov to not complain, I hate having a red X on my PRs haha
